crates/sui-core/src/authority_server.rs

```diff
@@ -827,6 +838,7 @@ impl ValidatorService {
         request: tonic::Request<HandleSoftBundleCertificatesRequestV3>,
     ) -> WrappedServiceResponse<HandleSoftBundleCertificatesResponseV3> {
         let epoch_store = self.state.load_epoch_store_one_call_per_task();
+        let client_addr = self.get_client_ip_addr(&request, &ClientIdSource::SocketAddr);
```

Comment:
Almost. Per our discussion this should be the folllowing:

```suggestion
        let client_addr = if self.client_id_source.is_none() {
            self.get_client_ip_addr(&request, &ClientIdSource::SocketAddr)
        } else {
            self.get_client_ip_addr(&request, client_id_source.unwrap())
        };
```